src/lib/gossip_net/libp2p.ml

```diff
@@ -60,12 +74,30 @@ let download_seed_peer_list uri =
   let%map contents = Cohttp_async.Body.to_string body in
   Mina_net2.Multiaddr.of_file_contents contents
 
-type publish_functions = { publish_v0 : Message.msg -> unit Deferred.t }
+type publish_functions =
+  { publish_v0 : Message.msg -> unit Deferred.t
+  ; publish_v1_block : Message.state_msg -> unit Deferred.t
+  ; publish_v1_tx : Message.transaction_pool_diff_msg -> unit Deferred.t
+  ; publish_v1_snark_work : Message.snark_pool_diff_msg -> unit Deferred.t
+  }
 
 let empty_publish_functions logger =
   { publish_v0 =
-      Fn.const
-      @@ Deferred.return ([%log warn] "Call of uninitialized publish_v0")
+      (fun _ ->
```

Comment:
Nit: should these be `failwith`s? It seems like a logic error if one of these gets called.

Reply:
Yeah, I think you're right